packages/odyssey-react-mui/src/useAutocomplete.tsx

```diff
@@ -0,0 +1,398 @@
+ */
+
+import {
+  createContext,
+  forwardRef,
+  HTMLAttributes,
+  ReactElement,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+// import _AutoSizer, {
+//   Props as AutoSizerProps,
+//   Size as AutoSizerSize,
+// } from "react-virtualized-auto-sizer";
+import { VariableSizeList, ListChildComponentProps } from "react-window";
+import { useTranslation } from "react-i18next";
+import styled from "@emotion/styled";
+import {
+  InputBase,
+  UseAutocompleteProps as MuiUseAutocompleteProps,
+  AutocompleteValue,
+  AutocompleteRenderInputParams,
+} from "@mui/material";
+
+import { AutocompleteProps } from "./Autocomplete";
+import { Field } from "./Field";
+import {
+  ComponentControlledState,
+  useInputValues,
+  getControlState,
+} from "./inputUtils";
+
+const ListboxContainer = styled.div({
+  width: "100%",
+  height: "100%",
+});
+
+// const AutoSizer = _AutoSizer as unknown as FC<AutoSizerProps>;
+type SetItemSize = (size: number) => void;
+
+export type UseAutocompleteProps<
+  OptionType,
+  HasMultipleChoices extends boolean | undefined,
+  IsCustomValueAllowed extends boolean | undefined,
+> = Pick<
+  AutocompleteProps<OptionType, HasMultipleChoices, IsCustomValueAllowed>,
+  | "ariaDescribedBy"
+  | "defaultValue"
+  | "errorMessage"
+  | "errorMessageList"
+  | "hasMultipleChoices"
+  | "hint"
+  | "HintLinkComponent"
+  | "inputValue"
+  | "isFullWidth"
+  | "isOptional"
+  | "isVirtualized"
+  | "label"
+  | "name"
+  | "onChange"
+  | "onInputChange"
+  | "testId"
+  | "value"
+>;
+
+export const useAutocomplete = <
+  OptionType,
+  HasMultipleChoices extends boolean | undefined,
+  IsCustomValueAllowed extends boolean | undefined,
+>({
+  ariaDescribedBy,
+  defaultValue,
+  errorMessage,
+  errorMessageList,
+  hasMultipleChoices,
+  hint,
+  HintLinkComponent,
+  inputValue,
+  isFullWidth,
+  isOptional,
+  isVirtualized: isVirtualizedProp,
+  label,
+  name: nameOverride,
+  onChange: onChangeProp,
+  onInputChange: onInputChangeProp,
+  testId,
+  value,
+}: UseAutocompleteProps<
+  OptionType,
+  HasMultipleChoices,
+  IsCustomValueAllowed
+>) => {
+  const { t } = useTranslation();
+  const controlledStateRef = useRef(
+    getControlState({
+      controlledValue: value,
+      uncontrolledValue: defaultValue,
+    }),
+  );
+
+  const isVirtualized = useRef(Boolean(isVirtualizedProp));
+
+  const defaultValueProp = useMemo<
+    | AutocompleteValue<
+        OptionType,
+        HasMultipleChoices,
+        undefined,
+        IsCustomValueAllowed
+      >
+    | undefined
+  >(() => {
+    if (hasMultipleChoices) {
+      if (value === undefined) {
+        return defaultValue;
+      }
+      return [] as AutocompleteValue<
+        OptionType,
+        HasMultipleChoices,
+        undefined,
+        IsCustomValueAllowed
+      >;
+    }
+    return value === undefined ? defaultValue : undefined;
+  }, [defaultValue, hasMultipleChoices, value]);
```

Comment:
I thought all this `hasMultipleChoices` logic was in MUI. If it's something we have to check for all over the place, this is why I suggest we separate them into separate components.

Is this the only place it's relevant?

Reply:
I think this is the only place we do any type of checking. But, I also feel like this could be a simpler. Again, I just moved this code into the hook for reusability.